spring-cloud-starter-stream-processor-tensorflow/src/main/java/org/springframework/cloud/stream/app/tensorflow/processor/TensorflowProcessorConfiguration.java

```diff
@@ -89,7 +93,23 @@ public Object evaluate(Message<?> input) {
 
 		Object outputData = tensorflowOutputConverter.convert(outputTensor, processorContext);
 
-		return outputData;
+		// put result in the header
+		if (properties.isSaveResultInHeader()) {
+			if (logger.isWarnEnabled()) {
+				if (input.getHeaders().containsKey(properties.getResultHeaderName())) {
+					logger.warn("Existing header [" + properties + "] will be overrided!");
+				}
+			}
+			return MessageBuilder
+					.withPayload(input.getPayload())
+					.setHeaderIfAbsent(properties.getResultHeaderName(), outputData)
+					.build();
```

Comment:
I've tried this but it fails the `LinearRegressionTensorflowProcessorIntegrationTests.java`. The `messageCollector.forChannel(channels.output()).poll()` starts to return the `MessageBuilder` instead of the expected payload.